cfme/common/provider.py

```diff
@@ -153,47 +150,153 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
         Returns:
             True if it was created, False if it already existed
         """
-        if check_existing and self.exists:
-            created = False
-        else:
-            created = True
-            logger.info('Setting up provider: %s', self.key)
-            navigate_to(self, 'Add')
-            fill(self.properties_form, self._form_mapping(True, hawkular=False, **self.__dict__))
-            for cred in self.credentials:
-                fill(self.credentials[cred].form, self.credentials[cred],
-                     validate=validate_credentials)
-            self._submit(cancel, self.add_provider_button)
-            if not cancel:
-                flash.assert_message_match('{} Providers "{}" was saved'.format(self.string_name,
-                                                                                self.name))
-        if validate_inventory:
-            self.validate()
+        from cfme.infrastructure.provider import InfraProvider
+        if self.one_of(InfraProvider):
+            if check_existing and self.exists:
+                created = False
+            else:
+                created = True
+
+                logger.info('Setting up provider: %s', self.key)
+                main_view = navigate_to(self, 'Add')
+
+                # filling main part of dialog
+                main_view.fill(self.view_value_mapping)
+
+                # filling endpoints
+                for endpoint in self.endpoints:
+                    try:
+                        endp_view = getattr(self.endpoints_form(parent=main_view), endpoint.name)
```

Comment:
**Optional**
Comment for getattr explaining its getting the view for the tab in the UI that matches the current endpoint. not entirely clear that's what the attribute is for from here, at least when first learning it.